### PR TITLE
DiscIO: Get rid of a few explicit deletes

### PR DESCRIPTION
**Pull request by lioncash**

### CODE REVIEW

File: Source/Core/DiscIO/VolumeWiiCrypted.cpp [hunk 1]
```diff
@@ -67,6 +64,7 @@ bool CVolumeWiiCrypted::Read(u64 _ReadOffset, u64 _Length, u8* _pBuffer, bool de
 
 	FileMon::FindFilename(_ReadOffset);
 
+	std::array<u8, s_block_total_size> read_buffer;
```
Comment (degasus) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: Please do not use such a big object on the stack.

**Comment by degasus** [comment]
Everything else looks good to me, nice cleanup.

File: Source/Core/DiscIO/VolumeWiiCrypted.cpp [hunk 2]
```diff
@@ -67,6 +63,7 @@ bool CVolumeWiiCrypted::Read(u64 _ReadOffset, u64 _Length, u8* _pBuffer, bool de
 
 	FileMon::FindFilename(_ReadOffset);
 
+	std::vector<u8> read_buffer(s_block_total_size);
```
Comment (JosJuice) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: I'd prefer having this as a member variable like before to avoid allocating this over and over.

Comment (lioncash) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: I wouldn't, since it'd be another thing to incorrectly mark as mutable.

Comment (degasus) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: It wastes 32 kB vs a mmap call. I'm not sure which one is better.

Comment (lioncash) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: I can assure you that in 2015, allocating 32KB will not kill performance.

**Comment by JosJuice** [comment]
LGTM.

**Pull request merged**